src/optimizer/optimizer.cpp

```diff
@@ -56,6 +58,14 @@ std::unique_ptr<OptimizeResult> Optimizer::BuildPlanTree(transaction::Transactio
     PlanGenerator generator(optimize_result->GetPlanMetaData());
     auto best_plan = ChooseBestPlan(txn, accessor, root_id, phys_properties, output_exprs, &generator);
     optimize_result->SetPlanNode(std::move(best_plan));
+
+    // Assign CTE Schema to each CTE Node
+    for (auto &table : context_->GetCTETables()) {
+      planner::CteScanPlanNode *leader = nullptr;
+      common::ManagedPointer<planner::CteScanPlanNode> ldr = common::ManagedPointer(leader);
+      ElectCTELeader(common::ManagedPointer(best_plan), table, &ldr);
```

Comment:
This could be due to my lack of C++ knowledge, but how can we use `best_plan` here after having moved it into `optimize_result` on line 60?

Reply:
@jkosh44 I think you're right.

Reply:
You should be fine to move `optimize_result->SetPlanNode(std::move(best_plan));` from line 60 to after this loop.

Reply:
Fixed